### PR TITLE
Create standardnotes-desktop.profile

### PR DESCRIPTION
**Pull request by veloute**

### CODE REVIEW

File: etc/standardnotes-desktop.profile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# This file is overwritten after every install/update
+# Persistent local customizations
+include /etc/firejail/standardnotes-desktop.local
+# Persistent global definitions
+include /etc/firejail/globals.local
+
+include /etc/firejail/disable-common.inc
+include /etc/firejail/disable-passwdmgr.inc
+include /etc/firejail/disable-programs.inc
+
```
Comment (Fred-Barclay) [inline etc/standardnotes-desktop.profile]: Can we try to add
```
include /etc/firejail/disable-devel.inc
include /etc/firejail/disable-interpreters.inc
```
As well?

The disable-interpreters one is a bit tricky to get right the first time. If something breaks, just post the terminal output here and we'll try to get it working. :smile:

File: etc/standardnotes-desktop.profile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+whitelist ${HOME}/Standard Notes Backups
+whitelist ${HOME}/.config/Standard Notes
+
+apparmor
+caps.drop all
+netfilter
```
Comment (Fred-Barclay) [inline etc/standardnotes-desktop.profile]: Can we add `machine-id`?

File: etc/standardnotes-desktop.profile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+nodbus
+nodvd
+nogroups
+nonewprivs
+noroot
```
Comment (Fred-Barclay) [inline etc/standardnotes-desktop.profile]: Does it need a microphone or play audio? If not, can you try `nosound` here?

File: etc/standardnotes-desktop.profile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+notv
+protocol unix,inet,inet6,netlink
+seccomp
```
Comment (Fred-Barclay) [inline etc/standardnotes-desktop.profile]: What happens if you add `tracelog`?

Comment (veloute) [reply etc/standardnotes-desktop.profile]: Everything but tracelog worked fine.

**Comment by veloute** [comment]
I did actually find one problem with
`include /etc/firejail/disable-interpreters.inc`
the process isn't killed when the window is closed.

File: etc/standardnotes-desktop.profile [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# Persistent global definitions
+include /etc/firejail/globals.local
+
+include /etc/firejail/disable-common.inc
+include /etc/firejail/disable-passwdmgr.inc
+include /etc/firejail/disable-programs.inc
+include /etc/firejail/disable-devel.inc
+include /etc/firejail/disable-interpreters.inc
+
+whitelist ${HOME}/Standard Notes Backups
```
Comment (SkewedZeppelin) [inline etc/standardnotes-desktop.profile]: mkdirs for these two paths?

File: etc/standardnotes-desktop.profile [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+machine-id
+nodbus
+nodvd
+nogroups
+nonewprivs
+noroot
+nosound
+notv
+protocol unix,inet,inet6,netlink
+seccom
```
Comment (SkewedZeppelin) [inline etc/standardnotes-desktop.profile]: seccom -> seccomp

can the following be added below?

```
disable-mnt
private-dev
private-tmp

noexec ${HOME}
noexec /tmp
```

and could you test this private-etc?

```
private-etc ca-certificates,fonts,host.conf,hostname,hosts,resolv.conf,ssl,pki,crypto-policies,xdg
```

File: etc/standardnotes-desktop.profile [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+# Persistent local customizations
+include /etc/firejail/standardnotes-desktop.local
+# Persistent global definitions
+include /etc/firejail/globals.local
+
+include /etc/firejail/disable-common.inc
+include /etc/firejail/disable-passwdmgr.inc
+include /etc/firejail/disable-programs.inc
+include /etc/firejail/disable-devel.inc
+include /etc/firejail/disable-interpreters.inc
```
Comment (SkewedZeppelin) [inline etc/standardnotes-desktop.profile]: alphabetize

File: etc/standardnotes-desktop.profile [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+whitelist ${HOME}/Standard Notes Backups
+whitelist ${HOME}/.config/Standard Notes
+
```
Comment (SkewedZeppelin) [inline etc/standardnotes-desktop.profile]: can `include /etc/firejail/whitelist-var-common.inc` be added?

**Comment by veloute** [comment]
Made all of the changes and they don't seem to cause any issues.

**Comment by SkewedZeppelin** [comment]
Merged, thanks!

**Comment by veloute** [comment]
It was and I couldn't get it working so it's not in the profile.

**Comment by veloute** [comment]
Seeing some problems that weren't showing up before.
`include /etc/firejail/disable-programs.inc`
seems to be the cause as it works fine when it's commented out.

Here's the log from running standardnotes-desktop in the terminal:
https://hastebin.com/raw/aroqogetef

**Comment by veloute** [comment]
https://github.com/netblue30/firejail/pull/2056

**Pull request merged**